sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -6001,6 +5985,12 @@ pi_result piEnqueueMemBufferCopy(pi_queue Queue, pi_mem SrcBuffer,
                                  pi_event *Event) {
   PI_ASSERT(SrcBuffer && DstBuffer, PI_INVALID_MEM_OBJECT);
   PI_ASSERT(Queue, PI_INVALID_QUEUE);
+
+  std::shared_lock SrcLock(SrcBuffer->Mutex, std::defer_lock);
+  std::unique_lock DstLock(DstBuffer->Mutex, std::defer_lock);
+  std::unique_lock QueueLock(Queue->Mutex, std::defer_lock);
+  std::scoped_lock LockAll(SrcLock, DstLock, QueueLock);
```

Comment:
This is shorter, although I am not sure if it reads better:
```suggestion
  std::shared_lock SrcLock(SrcBuffer->Mutex, std::defer_lock);
  std::scoped_lock LockAll(SrcLock, DstBuffer->Mutex, Queue->Mutex);
```

Reply:
Thanks, used shorter version.